src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentExtensions.cs

```diff
@@ -148,6 +148,7 @@ public static ContentElement Apply(this ContentElement contentElement, string na
             if (elementData != null)
             {
                 elementData.Merge(JObject.FromObject(element), JsonMergeSettings);
+                contentElement.Elements.Remove(name);
```

Comment:
This could be an issue if you have some state on the element that is not in the jobject (the reason we made the `Elements` map).

Reply:
Should we replace the existing element with the one passed as an argument instead? That might be less wrong.

Reply:
Replacing it seems a better option, Id assumed the merge applied to the ContentItem not the ContentElement so a complete replacement of the ContentElement makes sense.

I'll fix the other `Appy ` tomorrow.

Reply:
Then I will do it now ;) and you can approve/disapprove